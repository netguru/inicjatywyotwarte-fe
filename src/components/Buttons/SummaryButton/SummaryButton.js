import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'

const ButtonStyles = `
  display: block;
  text-decoration: none;
  text-align: center;
  color: inherit;
  border-radius: 50px;
  padding: 0.7rem 1rem;
  font-weight: 500;
  font-size: 0.8rem;
  margin: 2rem auto;
  width: 10rem;
  cursor: pointer;
`

const StyledLink = styled(Link).attrs(() => ({ "data-cy": "cookies-banner-accept" }))`
  ${ButtonStyles}
  border: 2px solid ${props => props.theme.modalColorVeryLight};
  background: ${props => props.theme.lightBackgroundColor};
  transition: ${props => props.theme.buttonTransition};
  
  &:hover {
    box-shadow: ${props => props.theme.shadowButtonHover};
  }
`
StyledLink.displayName = 'SummaryLink';

const StyledButton = styled.div.attrs(() => ({ "data-cy": "cookies-banner-accept" }))`
  ${ButtonStyles}
  border: 2px solid ${props => props.theme.modalColorVeryLight};
  background: ${props => props.theme.lightBackgroundColor};
  transition: ${props => props.theme.buttonTransition};
  
  &:hover {
    box-shadow: ${props => props.theme.shadowButtonHover};
  }
`
StyledButton.displayName = 'SummaryButton';

const SummaryButton = ({ to = '', onClick, children, className }) => {
  if (!to && onClick) {
    return (
      <StyledButton className={className} onClick={onClick}>
        {children}
      </StyledButton>
    )
  }
  return (
    <StyledLink className={className} to={to}>
      {children}
    </StyledLink>
  )
}

export default SummaryButton
