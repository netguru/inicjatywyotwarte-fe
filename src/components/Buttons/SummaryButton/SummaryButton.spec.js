import React from 'react'
import SummaryButton from './SummaryButton'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  onClick: jest.fn('mocked-function'),
  children: <p>Child</p>,
  className: 'mocked-class-name'
}

describe('SummaryButton', () => {
  test('renders button', () => {
    const wrapper = shallow(<SummaryButton {...baseProps} />);
    expect(wrapper.find('SummaryButton').length).toEqual(1);
    expect(wrapper.find('[className="mocked-class-name"]').length).toEqual(1);
    expect(wrapper.children().first().equals(<p>Child</p>)).toEqual(true);
  });

  test('renders link when to attr in props', () => {
    const propsWithToAttribute = {
      ...baseProps,
      to: 'mocked-to'
    }
    const wrapper = shallow(<SummaryButton {...propsWithToAttribute} />);
    expect(wrapper.find('SummaryLink[className="mocked-class-name"][to="mocked-to"]').length).toEqual(1);
    expect(wrapper.children().first().equals(<p>Child</p>)).toEqual(true);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<SummaryButton/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


