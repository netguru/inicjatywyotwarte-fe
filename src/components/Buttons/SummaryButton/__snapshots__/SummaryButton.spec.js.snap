// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryButton match snapshot 1`] = `
<SummaryLink
  to=""
/>
`;
